gse-util/src/main/java/com/powsybl/gse/util/NewScriptButton.java

```diff
@@ -85,14 +78,10 @@ public static Button create(ProjectNode node, ObjectProperty<ModificationScript>
         pane.add(folderSelectionPane.getTextField(), 1, 1, 2, 1);
         pane.add(folderSelectionPane.getButton(), 3, 1);
         pane.add(nameTextField.getFileAlreadyExistsLabel(), 0, 2, 4, 1);
-        dialog.getDialogPane().setContent(pane);
         BooleanBinding okProperty = folderSelectionPane.nodeProperty().isNotNull()
                                                        .and(nameTextField.okProperty());
-        button.disableProperty().bind(okProperty.not());
-        dialog.setResizable(true);
-        dialog.setResultConverter(buttonType -> buttonType == ButtonType.OK ? Boolean.TRUE : Boolean.FALSE);
         Platform.runLater(nameTextField.getInputField()::requestFocus);
-        return dialog;
+        return new GseDialog<>(RESOURCE_BUNDLE.getString("DialogTitle"), pane, window, okProperty.not(), buttonType -> buttonType == ButtonType.OK ? Boolean.TRUE : Boolean.FALSE);
```

Comment:
Extract the lambda used as a Callback to a variable to improve readability.

Reply:
updated